pkg/cloudprovider/plugins/aws/aws_ec2.go

```diff
@@ -396,8 +373,19 @@ func (ec2Cfg *ec2ServiceConfig) GetVpcInventory() map[string]*runtimev1alpha1.Vp
 			strings.ToLower(ec2Cfg.credentials.region), managed)
 		vpcMap[strings.ToLower(*vpc.VpcId)] = vpcObj
 	}
+	return vpcMap
+}
 
-	awsPluginLogger().V(1).Info("Cached vpcs", "account", ec2Cfg.accountNamespacedName, "vpc objects", len(vpcMap))
+// GetCloudInventory fetches VM and VPC inventory from stored snapshot and converts from cloud format to internal format.
+func (ec2Cfg *ec2ServiceConfig) GetCloudInventory() *nephetypes.CloudInventory {
+	cloudInventory := nephetypes.CloudInventory{
+		VmMap:  map[types.NamespacedName]map[string]*runtimev1alpha1.VirtualMachine{},
+		VpcMap: map[string]*runtimev1alpha1.Vpc{},
```

Comment:
Why we need map for VM and VPC. I get [selector] to <vm>. But why <vm> is a map. It can be just a slice right?

Reply:
This needs to be a map for the deletion logic in BuildVmCache/BuildVpcCache, we need to find out if a vm/vpc in inventory is present in discovered vms from cloud.